railties/lib/rails/commands/rake_proxy.rb

```diff
@@ -0,0 +1,38 @@
+require 'rake'
+
+module Rails
+  module RakeProxy
+
+    RAKE_TASKS_HELP_MESSAGE = <<-EOT
+In addition to those, you can now run the rake tasks as rails commands:
+
+    EOT
+
+    def write_rake_tasks_help_message
+      puts RAKE_TASKS_HELP_MESSAGE
+    end
+
+    def write_rake_tasks
+      width = rake_tasks.map { |t| t.name_with_args.length }.max || 10
+      rake_tasks.each do |t|
+        printf("#{Rake.application.name} %-#{width}s  # %s\n", t.name_with_args, t.comment)
+      end
+    end
+
+    def rake_tasks
+      require_application_and_environment!
+      Rake::TaskManager.record_task_metadata = true
+      Rake.application.instance_variable_set(:@name, 'rails')
+      Rails.application.load_tasks
+      @rake_tasks ||= Rake.application.tasks.select { |t| t.comment }
```

Comment:
Seems like the ivar should wrap the whole method, otherwise we're loading tasks twice.

Also do `select(&:comment)`